rust/benches/file_backend.rs

```diff
@@ -245,24 +245,55 @@ pub fn backend_open_fns() -> impl Iterator<Item = BackendOpenFn> {
 }
 
 fn file_backend_benchmark_matrix(c: &mut Criterion) {
-    let plot_config = PlotConfiguration::default().summary_scale(criterion::AxisScale::Logarithmic);
+    let meminfo = fs::read_to_string("/proc/meminfo").unwrap();
+    let mem_total = meminfo
+        .lines()
+        .find(|l| l.starts_with("MemTotal:"))
+        .unwrap();
+    let memory_kb: u64 = mem_total
+        .strip_prefix("MemTotal:")
+        .unwrap()
+        .strip_suffix("kB")
+        .unwrap()
+        .trim()
+        .parse()
+        .unwrap();
+
+    // Because we do not use the default test harness, cfg!(test) is true also when running with
+    // `cargo bench`, but cfg(bench) is not. To detect if we are running the benchmarks as tests, we
+    // use cfg!(debug_assertions) as a proxy because by default tests are run in debug mode and
+    // benchmarks are not.
+    let target_file_size = if cfg!(debug_assertions) {
+        100_000
+    } else {
+        memory_kb * 1024 * 4
```

Comment:
```suggestion
        let size = memory_kb * 1024 * FILE_SIZE_MEMORY_MULTIPLIER;
        eprintln!("Using benchmark file of size {} * main memory size ({}) GiB) = {} GiB to limit effects of OS page cache", FILE_SIZE_MEMORY_MULTIPLIER, memory_kb / 1024 / 1024, size / ONE_GB);
        size
```

We should give some indication that we are about to write a huge file somewhere, and why.